### ISSUE TITLE
Stock Albatross Hardpoints seem bugged.

### ISSUE THREAD

**Issue by Lineth** [endless-sky/endless-sky#3121]
So when you buy a new Albatross, it will weapons that do not exist on it in space, and if you strip it of its weapons and install new ones, the weapons installed in place of these 'ghost' weapons will not fire. Entering the ship info panel seems to fix this, however.

**Comment by tehhowch** [comment]
This looks fine to me: https://youtu.be/gUcdYSNWoVA

**Comment by Lineth** [comment]
Hm. Seems to be the result of any plugin that modifies the Albatross' outfit values. Without them, it behaves as normal, but with a buff to its stat,s it reports every hardpoint being filled with an inhibitor cannon and four thrasher cannons and a point defense turret. I have no idea what's causing this, as it seems to be the only ship that does it.

**Comment by tehhowch** [comment]
This sounds like an improperly specified ship definition issue and could possibly be rectified by #2959 

Please post the exact plugin code that is used to modify the Albatross.

**Comment by Lineth** [comment]
`ship "Albatross"
	attributes
		"engine capacity" 172
		"outfit space" 750
		"weapon capacity" 302`

**Comment by tehhowch** [comment]
#2901 and #2959 appear very relevant

**Comment by tehhowch** [comment]
Here's the bug in action, running fefef5babc83ef008e8049d6f719e2c1ed8cb0e4 :
https://youtu.be/7DS15d2WO6g

All installed Inhibitor Cannons are sold, as are all Thrasher Cannons. 7 Beam Lasers are then installed. The ship is then launched and fires... but what fires is 6 beam lasers and 1 inhibitor cannon.
Viewing the Ship Info panel then shows this screen:
![image](https://user-images.githubusercontent.com/20871346/31560398-bbb96374-b019-11e7-931e-5f637d9976b4.png)
Landing, changing the hardpoint assignment, and then re-launching does not help:
https://youtu.be/iorgWjZJcYs
This final video is using `ship "Albatross" "Albatross"` as well. What is interesting is the single shot that fires from the thrasher turret.

In the end, this comes down to an unsupported override of the ship model specification.
With #2959 you could do
```
ship "Albatross"
  add attributes
    "engine capacity" <desired> - <current>
    "outfit space" <desired> - <current>
    "weapon capacity" <desired> - <current>
```
to get the desired effect.

**Comment by Lineth** [comment]
Should I rename this issue, as it seems emblematic of a wider issue? Sorry if silly question.

**Comment by tehhowch** [comment]
Your specific use case is resolved with #2959, but the underlying issue is not (it is only made more difficult to encounter due to the existence of supported attribute modification methods).

I would let endless-sky pick the appropriate title for the underlying bug that remains. Given that the ability to add to attributes and not redefine the base set now exists, this may just get closed with a milestone.

**Comment by endless-sky** [comment]
Closing because this is a bug in the plugin, not the game itself.
